inst/include/dust/random/multinomial.hpp

```diff
@@ -26,23 +26,24 @@ namespace random {
 // with use from odin code.
 template <typename real_type, typename rng_state_type,
           typename T, typename U>
+__host__ __device__
```

Comment:
Do `prob` and `ret` need any special considerations? I suppose that they be benefit from interleaving, but maybe that could just be left as a comment for now.

Reply:
I think the way it currently is written, prob needs work as it currently requires a `real_type*`, but `ret` will automatically cope with an interleaved type. There's a linked pr that you might peek at where I think it's more obvious

Reply:
Ah, no this is just `const T&` for prob and `U&` for prob an ret respectively. So that means it should do the right thing.  Here's the other PR : https://github.com/mrc-ide/dust-random-bench/pull/1 we should have a look there (either in this PR or another) to wire this up properly.